src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceChannelFactory.cs

```diff
@@ -450,8 +482,7 @@ protected override void OnEndClose(IAsyncResult result)
 
             protected internal override Task OnCloseAsync(TimeSpan timeout)
```

Comment:
I'm not saying you need to fix it in this iteration as your change does address the specific problem that's happening, but if this is checked in as-is we should open an issue to track that we have multiple implementations and to clean it up to a single implementation.

---
In reply to: [60986686](https://github.com/dotnet/wcf/pull/1071#discussion_r60986686) [](ancestors = 60986686)